### DISCUSSION TITLE
Stuck indexing local files

### DISCUSSION THREAD

**Discussion by chrisrhymes** [tinacms/tinacms#4206] [Help] [answered]
I am trying to setup tinacms to work with an existing Jekyll site. I have followed the framework specific guide for [Jekyll](https://tina.io/docs/frameworks/jekyll/) and updated the config.ts file but it is stuck "Indexing local files" in the terminal window and "Please wait, Tina is loading data..." in the browser window when viewing "Posts". 

I have left this running for several minutes and it doesn't stop. 

Here is my configuration: 

```js
import { defineConfig } from "tinacms";

// Your hosting provider likely exposes this as an environment variable
const branch = process.env.HEAD || process.env.VERCEL_GIT_COMMIT_REF || "main";

export default defineConfig({
  branch,
  clientId: null, // Get this from tina.io
  token: null, // Get this from tina.io

  build: {
    outputFolder: "admin",
    publicFolder: "./",
  },
  media: {
    tina: {
      mediaRoot: "assets/images",
      publicFolder: "./",
    },
  },
  schema: {
    collections: [
      {
        name: "post",
        label: "Posts",
        path: "_posts",
        fields: [
          {
            type: "string",
            name: "title",
            label: "Title",
            isTitle: true,
            required: true,
          },
          {
            type: "string",
            name: "layout",
            label: "Layout",
            required: true,
          },
          {
            type: "rich-text",
            name: "body",
            label: "Body",
            isBody: true,
          },
        ],
      },
    ],
  },
});
```

**Comment by kldavis4** [comment]
do you have a git repository with a reproduction that we can look at?

**Reply by chrisrhymes** [reply]
Hi, I have created an example repo here that replicates the same issue: https://github.com/chrisrhymes/jekyll-tinacms-demo

I tried `npx tinacms dev -c "jekyll serve"` and `npx tinacms dev -c "bundle exec jekyll serve"` but both have the same issue.

**Reply by kldavis4** [reply]
I'm not currently able to replicate the issue. See this screen recording: https://www.loom.com/share/afa19cc3827c45c4bc725877cbf6623e?sid=e7c9afad-e2bb-410d-bee4-6751531ecb33 - let me know if there is something else I need to do to replicate the issue

**Reply by chrisrhymes** [reply]
Thank you for taking a look at this for me. I tried running the tinacms and serving the root directory separately but it had the same issue where it just says `Please wait, Tina is loading data...`.  

I was wondering if you had a link to an example repository with TinaCMS and Jekyll that I could take a look at to see if there are any obvious differences? 

After a bit more investigation I looked at the network tab when visiting `/admin/#/collections/post/` and there seems to be a request sent to `http://localhost:4001/graphql` that shows as pending and doesn't return a response. That was the only thing I could see not working as it should. This is the payload:

```graphql
{
    "query": "#graphql\n      query($collection: String!, $includeDocuments: Boolean!, $sort: String,  $limit: Float, $after: String, $filter: DocumentFilter, $folder: String){\n        collection(collection: $collection){\n          name\n          label\n          format\n          templates\n          documents(sort: $sort, after: $after, first: $limit, filter: $filter, folder: $folder) @include(if: $includeDocuments) {\n            totalCount\n            pageInfo {\n              hasPreviousPage\n              hasNextPage\n              startCursor\n              endCursor\n            }\n            edges {\n              node {\n                __typename\n                ... on Folder {\n                    name\n                    path\n                }\n                ... on Document {\n                  _sys {\n                    title\n                    template\n                    breadcrumbs\n                    path\n                    basename\n                    relativePath\n                    filename\n                    extension\n                  }\n                }\n              }\n            }\n          }\n        }\n      }",
    "variables": {
        "collection": "post",
        "includeDocuments": true,
        "folder": "",
        "sort": "title",
        "limit": 50,
        "after": "",
        "filter": null
    }
}
```

I also tried running `npx tinacms audit` and it returned the following error:

```
npx tinacms audit
Starting Tina Audit
Indexing local files ⠋ModuleError: Connection to leader lost
    at TinaLevelClient.[abortRequests] (/code/test-jekyll-tina/node_modules/many-level/guest.js:194:20)
```

**Reply by chrisrhymes** [reply, marked as answer]
Right, sorted it. I was running an old version of node and npm. Updated to node 18 (from 16) and it is now working as expected. 

I also had to update the post extension from `.markdown` to `.md` for it to find the existing post. 

Thanks for your help @kldavis4
